crates/ruff/src/args.rs

```diff
@@ -291,7 +306,14 @@ pub struct CheckCommand {
     #[arg(short, long, env = "RUFF_NO_CACHE", help_heading = "Miscellaneous")]
     pub no_cache: bool,
     /// Ignore all configuration files.
-    #[arg(long, conflicts_with = "config", help_heading = "Miscellaneous")]
+    //
+    // Note: We can't mark this as conflicting with `--config` here
+    // as `--config` can be used for specifying config overrides as well as config files.
+    // Specifying a config file conflicts with `--isolated`;
+    // specifying a config override does not.
```

Comment:
Why does specifying a config file conflict with `--isolated`? Shouldn't `--isolated` just ignore any _discovered_ config files? I don't see any problem with respecting config files that are explicitly passed.

Reply:
I don't have a strong opinion here; this just preserves the pre-existing behaviour where it was an error to pass `--config <config file>` and also pass `--isolated` on the CLI. I added the comment because we now have to generate that error manually rather than using `clap` to generate it, so it's less obvious from the definitions here that this would be the case

Reply:
I don't see a reason to ban the behavior. Our config file inference is really complicated and passing in `--isolated` to disable that doesn't seem like it should prevent you from passing in a single configuration file.

Curious for @charliermarsh's thoughts though because I do not have context on the origin of these options.

Reply:
Another question is... does `--config=...` imply isolated? or does it apply that configuration file on top of any discovered configuration files?

Reply:
> or does it apply that configuration file on top of any discovered configuration files?

I _believe_ this is the current behaviour

Reply:
I think allowing it _with_ `--isolated` makes a lot of sense then.

Reply:
I'm pretty sure `--config` "implies" `--isolated`, in that if you use `--config`, we ignore any discovered configuration files.

Reply:
I think it's fine to allow them to be used together (even if that differs from the previous conflict settings).

Reply:
So this PR is currently going out of its way to _preserve_ the behaviour on `main`, which is that `--config <config_file>` and `--isolated` aren't allowed to be specified simultaneously. On `main` we just use `clap` to forbid them both from being specified at once, but with this PR it ends up being a bunch of extra code to achieve that prohibition, because we _do_ want to allow `--config "<setting> = <value>"` in combination with `--isolated`.

The question is whether this PR _should_ be going out of its way to preserve the behaviour on `main` (which is what it currently does), or if it would actually be _desirable_ to change things so that it's now legal to specify `--isolated` in combination with `--config <config_file>`.

Reply:
My inclination is to keep this PR how it is, and consider any possible behaviour changes relating to this in a separate issue/PR, if at all. There's not _that_ much extra code being added here to maintain the previous behaviour, and it should be pretty easy to rip it out if we _do_ want to change the behaviour